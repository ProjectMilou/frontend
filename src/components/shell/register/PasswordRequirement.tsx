import React from 'react';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import RadioButtonUncheckedIcon from '@material-ui/icons/RadioButtonUnchecked';
import { createStyles, makeStyles } from '@material-ui/core';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import { Theme } from '@material-ui/core/styles';

interface PasswordRequirementProps {
  done: boolean;
  text: string;
}

const useStyles = makeStyles<Theme, PasswordRequirementProps>((theme) =>
  createStyles({
    checked: {
      height: theme.spacing(1.5),
      color: theme.palette.success.main
    },
    unchecked: {
      height: theme.spacing(1.5),
    },
    name: {
      fontSize: '12px',
      color: ({done}) => (done ? "grey": ""),
      textDecoration: ({done}) => (done ? "line-through": ""),
    },
  })
);

const PasswordRequirement: React.FC<PasswordRequirementProps> = (props) => {
  const { done, text } = props;
  const { checked, unchecked, name } = useStyles(props);
  return (
    <Box display="flex" alignItems="center">
      {done ? (
        <CheckCircleIcon className={checked} />
      ) : (
        <RadioButtonUncheckedIcon className={unchecked} />
      )}
      <Typography className={name}>{text}</Typography>
    </Box>
  );
};

export default PasswordRequirement;
