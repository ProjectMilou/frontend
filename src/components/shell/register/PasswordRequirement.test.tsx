import { render } from '@testing-library/react';
import * as React from 'react';
import PasswordRequirement, {
  PasswordRequirementProps,
} from './PasswordRequirement';

describe('PasswordRequirement', () => {
  const renderPasswordRequirement = (props: PasswordRequirementProps) =>
    render(<PasswordRequirement {...props} />);

  test('should render valid requirement', async () => {
    const component = renderPasswordRequirement({ done: true, text: 'text' });
    expect(component).toMatchSnapshot();
  });
  test('should render invalid requirement', async () => {
    const component = renderPasswordRequirement({ done: false, text: 'text' });
    expect(component).toMatchSnapshot();
  });
});
