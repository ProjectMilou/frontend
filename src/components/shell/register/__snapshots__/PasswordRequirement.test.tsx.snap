// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordRequirement should render invalid requirement 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-10"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-unchecked-7"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-name-8 makeStyles-name-9 MuiTypography-body1"
        >
          text
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-10"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-unchecked-7"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-name-8 makeStyles-name-9 MuiTypography-body1"
      >
        text
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PasswordRequirement should render valid requirement 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-checked-1"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-name-3 makeStyles-name-4 MuiTypography-body1"
        >
          text
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-checked-1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <p
        class="MuiTypography-root makeStyles-name-3 makeStyles-name-4 MuiTypography-body1"
      >
        text
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
